spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewriteManifestsAction.java

```diff
@@ -956,6 +956,62 @@ public void testRewriteLargeDeleteManifestsPartitionedTable() throws IOException
     assertThat(deleteManifests).hasSizeGreaterThanOrEqualTo(2);
   }
 
+  @TestTemplate
+  public void testUpgradeToV3() throws IOException {
```

Comment:
I feel a little awkward about the scope of this test.  It's testing upgrade and manifest rewrite, but there are specific aspects that are included.

I feel like there are more scenarios to cover here (e.g. writing a DV to an existing file that has positional delete after the upgrade.).  I would say either refine the name to be more explicit (e.g. `testUpgradeToV3WithExistingV2Deletes`) so that we can add more test cases for the upgrade path.

Reply:
I agree the name is too generic. I did cover adding a DV to an existing file with positional delete in a previous PR [here](https://github.com/apache/iceberg/pull/11467/files#diff-f7cf300e48afd586e158fba64c058d30d14d37b3e96739d6cca23758ee444593R633)).

Reply:
There is also an extra test in `TestRowDelta`. This one was for the action that rewrites manifests, but I renamed both.